pkg/aggregator/internal/limiter/limiter.go

```diff
@@ -27,10 +29,17 @@ type Limiter struct {
 	keyTagName        string
 	telemetryTagNames []string
 	limit             int
+	global            int // global limit
+	current           int // sum(usage[*].current)
 	usage             map[string]*entry
+
+	// epoch, maxAge and lastSeen ensure eventual removal of entries that created an entry, but were
+	// never able to create contexts due to the global limit.
+	epoch  int
+	maxAge int
 }
 
-// New returns a new instance of limiter.
+// New returns a limiter with a per-key limit.
```

Comment:
```suggestion
// New returns a limiter with a per-sender limit.
```

Just trying to stay a bit consistent, we use `key` `origin` and `instance` in different places all to refer to the same concept.